homeassistant/components/tuya/config_flow.py

```diff
@@ -82,7 +83,11 @@ def _try_login(user_input):
         if project_type == ProjectType.INDUSTY_SOLUTIONS:
             response = api.login(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
         else:
-            api.endpoint = TUYA_ENDPOINT_BASE
+            if user_input[CONF_COUNTRY_CODE] in COUNTRY_CODE_CHINA:
+                api.endpoint = TUYA_ENDPOINT_BASE
+            else:
+                api.endpoint = TUYA_ENDPOINT_OTHER
```

Comment:
What about the other regions?
Should everyone use America or china?
For me in Europe, "https://openapi.tuyaeu.com" worked great
I wonder if it makes sense to add `vol.Required(CONF_ENDPOINT): vol.In(TUYA_ENDPOINT),` to this schema as well?

Reply:
Yeah I mentioned I was thinking the same on my PR.

Certainly the US endpoint should work for all but China, but makes sense to use closest where possible.

Reply:
For now, it only limit the cross-region between China and other countries. For countries outside China,  no cross-region limitation. And will select the correct account region automatically.